src/main/java/com/direwolf20/buildinggadgets/common/building/view/MapBackedBuildView.java

```diff
@@ -0,0 +1,112 @@
+package com.direwolf20.buildinggadgets.common.building.view;
+
+import com.direwolf20.buildinggadgets.common.building.BlockData;
+import com.direwolf20.buildinggadgets.common.building.PlacementTarget;
+import com.direwolf20.buildinggadgets.common.building.Region;
+import com.direwolf20.buildinggadgets.common.util.spliterator.MappingSpliterator;
+import com.google.common.collect.ImmutableMap;
+import net.minecraft.util.math.BlockPos;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.Spliterator;
+import java.util.function.Function;
+
+/**
+ * A simple {@link IBuildView} backed by a {@link Map Map<BlockPos, BlockData>}. {@link PlacementTarget PlacementTargets} will be created
+ * lazily when iterating over this {@link IBuildView}. You can supply this with a mutable {@link Map} via {@link #createUnsafe(IBuildContext, Map, Region)}
+ * for efficiency reasons, note however that you will encounter undefined behaviour if the {@link Map} is modified after this {@link IBuildView} was
+ * created.
+ * <p>
+ * Notice that closing has no effect on this object, and therefore also isn't required.
+ */
+public final class MapBackedBuildView implements IBuildView {
+    private Map<BlockPos, BlockData> map;
+    private Region boundingBox;
+    private BlockPos translation;
+    private IBuildContext context;
+
+    public static <T> MapBackedBuildView ofIterable(IBuildContext context, Iterable<T> iterable, Function<? super T, ? extends BlockPos> keyExtractor, Function<? super T, BlockData> dataExtractor) {
```

Comment:
It's a Map, not a list. And java isn't F#which uses lists for it's maps. I don't like the term PosititonalBuildView, because in contrast to MapBackedBuildView it doesn't directly tell you what it is.

Reply:
Most languages that have Lists and maps are just mapped lists but still

Reply:
Why do they kill their Performance like that? I can see that writing a HashMap might be tedious, but the algorithms are already designed...

Anyway as you are responsible for code readability I'm going to change it by your Request, still don't like it that much though.